### PR TITLE
Support 'throw' as a function attribute

### PR DESCRIPTION
**Pull request by Hackerpilot**
See https://github.com/dlang-community/libdparse/issues/423

### CODE REVIEW

**Comment by Hackerpilot** [comment]
The example file produces the correct AST, but this could probably use a bit more testing.

**Comment by Hackerpilot** [comment]
After thinking about it a bit more, I think it's better to get this change merged in so that the various tools that depend on libdparse can support this syntax when the next compiler release comes out. Testing that the parser produces the correct AST (See https://github.com/dlang-community/libdparse/issues/143) is a much bigger issue and makes sense to do separately from this.

File: test/pass_files/throwattribute.d [hunk 1]
```diff
@@ -0,0 +1,10 @@
+void doStuff() throw {
+	throw void doThings()
```
Comment (AndrejMitrovic) [inline test/pass_files/throwattribute.d:2]: What about a fail case for `void foo (throw int x)`? Or does it not matter / it's already handled?

Comment (Hackerpilot) [reply test/pass_files/throwattribute.d:2]: It doesn't parse.

Comment (Hackerpilot) [reply test/pass_files/throwattribute.d:2]: And, actually, putting a test case for that into the fail_files directory was really easy, so I did it.

**Pull request merged**